azurerm/resource_arm_api_management.go

```diff
@@ -0,0 +1,773 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/apimanagement/mgmt/2017-03-01/apimanagement"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmApiManagementService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmApiManagementServiceCreateUpdate,
+		Read:   resourceArmApiManagementServiceRead,
+		Update: resourceArmApiManagementServiceCreateUpdate,
+		Delete: resourceArmApiManagementDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateApiManagementName,
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"location": locationSchema(),
+
+			"publisher_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateApiManagementPublisherName,
+			},
+
+			"publisher_email": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateApiManagementPublisherEmail,
+			},
+
+			"sku": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  string(apimanagement.SkuTypeDeveloper),
+							ValidateFunc: validation.StringInSlice([]string{
+								string(apimanagement.SkuTypeDeveloper),
+								string(apimanagement.SkuTypeBasic),
+								string(apimanagement.SkuTypeStandard),
+								string(apimanagement.SkuTypePremium),
+							}, true),
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+						},
+						"capacity": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  1,
+						},
```

Comment:
minor, could we validate this? I presume `IntAtLeast(0)` would be a good starting point